metrics.go

```diff
@@ -21,21 +21,38 @@ var (
 
 func sanitizeValue(label string, value interface{}) string {
 
-	// Convert to a string, no matter what underlying type it is
-	var labelValue string
-	if value != nil {
-		labelValue = fmt.Sprintf("%v", value)
+	if value == nil || value == "" || value == "-" {
+		return ""
 	}
 
+	// Convert to a string, no matter what underlying type it is
+	labelValue := fmt.Sprintf("%v", value)
+	labelValue = strings.TrimSpace(labelValue)
+
 	switch label {
 	case "content_type":
-		labelValue = strings.Split(labelValue, ";")[0]
+		labelValue = strings.ToLower(labelValue)
+		if strings.HasPrefix(labelValue, "image") {
+			labelValue = "image"
+		} else if strings.HasPrefix(labelValue, "text/html") {
+			labelValue = "html"
+		} else if strings.HasPrefix(labelValue, "text/css") {
+			labelValue = "css"
+		} else if strings.Contains(labelValue, "javascript") {
+			labelValue = "javascript"
+		} else {
+			labelValue = "other"
+		}
 	case "hostname":
 		labelValue = strings.Split(labelValue, ":")[0]
+		labelValue = strings.ToLower(labelValue)
+	case "status":
+		statusInt, err := strconv.Atoi(labelValue)
+		if err != nil || statusInt < 100 || statusInt > 599 {
```

Comment:
In logstash we prune status codes even more aggressively down to just 67 possibilities:
https://github.com/section-io/section.aperture/blob/master/puppet/squixa_v3_logs_indexer/files/etc-logstash-processor-b-conf.d/10-shared-ingress.conf#L121-L137
Considered?